src/parser/Parser.ts

```diff
@@ -992,36 +994,61 @@ export class Parser {
         let asToken: Token;
         let typeExpression: TypeExpression;
 
-        //look for `as SOME_TYPE`
-        if (this.check(TokenKind.As)) {
-            this.warnIfNotBrighterScriptMode('typed assignment');
+        if (allowTypedAssignment) {
+            //look for `as SOME_TYPE`
+            if (this.check(TokenKind.As)) {
+                this.warnIfNotBrighterScriptMode('typed assignment');
 
-            [asToken, typeExpression] = this.consumeAsTokenAndTypeExpression();
+                [asToken, typeExpression] = this.consumeAsTokenAndTypeExpression();
+            }
         }
 
         let operator = this.consume(
-            DiagnosticMessages.expectedOperatorAfterIdentifier(allowedAssignmentOperators, name.text),
-            ...allowedAssignmentOperators
+            DiagnosticMessages.expectedOperatorAfterIdentifier([TokenKind.Equal], name.text),
+            ...[TokenKind.Equal]
         );
         let value = this.expression();
 
-        let result: AssignmentStatement;
+        let result = new AssignmentStatement({ equals: operator, name: name, value: value, as: asToken, typeExpression: typeExpression });
+
+        /*AssignmentStatement | AugmentedAssignmentStatement;
         if (operator.kind === TokenKind.Equal) {
-            result = new AssignmentStatement({ equals: operator, name: name, value: value, as: asToken, typeExpression: typeExpression });
+            result =
         } else {
             const nameExpression = new VariableExpression({ name: name });
-            result = new AssignmentStatement({
-                equals: operator,
-                name: name,
-                value: new BinaryExpression({
-                    left: nameExpression,
-                    operator: operator,
-                    right: value
-                }),
-                as: asToken,
-                typeExpression: typeExpression
+            result = new AugmentedAssignmentStatement({
+                item: nameExpression,
+                operator: operator,
+                value: value
             });
         }
+*/
+        return result;
+    }
+
+    private augmentedAssignment(): AugmentedAssignmentStatement {
+        let name = this.advance() as Identifier;
```

Comment:
What about dotted get and indexed get? Shouldn't it be `this.expression()`?

```
array[0] += 1
alpha.beta.charlie += 2
```

Reply:
I'll change it so all uses of `+=`, '-=`, etc... will be Augmented Assignments

Reply:
done